web/src/message_list.js

```diff
@@ -182,6 +183,11 @@ export class MessageList {
             this.select_id(first_unread_message_id, {then_scroll: true, use_closest: true});
         }
 
+        // Rebuild message list, since we might need to shuffle around the participant users.
+        if (narrow_state.stream_sub() && narrow_state.topic()) {
+            activity_ui.build_user_sidebar();
+        }
```

Comment:
This should only apply if the message list equals `message_lists.current`; we support background maintaining others, and updating those should not trigger a rerender. Likely reproducibly causes duplicate rerenderes if you have "combined feed" as your home view.